src/main/java/io/jenkins/plugins/sprp/PipelineSnippetGenerator.java

```diff
@@ -349,17 +333,39 @@ public String getPublishArtifactStage(ArtifactPublishingConfig config,
 
         StringBuilder snippet = new StringBuilder("stage('Publish Artifact') {\n");
 
-        snippet.append("\tsteps {\n");
-        snippet.append("\t\t" + "withCredentials([file(credentialsId: '").append(config.getCredentialId()).append("', variable: 'FILE')]) {\n");
+        snippet.append("steps {\n");
+        snippet.append("" + "withCredentials([file(credentialsId: '").append(config.getCredentialId()).append("', variable: 'FILE')]) {\n");
 
         for(HashMap<String, String> artifact: publishArtifacts){
-            snippet.append("\t\t\tsh 'scp -i $FILE ").append(artifact.get("from")).append(" ").append(config.getUser()).append("@").append(config.getHost()).append(":").append(artifact.get("to")).append("'\n");
+            snippet.append("sh 'scp -i $FILE ").append(artifact.get("from")).append(" ").append(config.getUser()).append("@").append(config.getHost()).append(":").append(artifact.get("to")).append("'\n");
         }
 
-        snippet.append("\t\t}\n");
-        snippet.append("\t}\n");
+        snippet.append("}\n");
+        snippet.append("}\n");
         snippet.append("}\n");
 
         return snippet.toString();
     }
+
+    public StringBuilder autoAddTabs(StringBuilder pipelineScript){
```

Comment:
Does this take a long time to run?  Originally I thought it would be easier to create a helper method that would automatically append the number of tabs at initial `snippet.append` time.  Like: `snippet.append(tab("steps{\n")).  That could take longer to execute, I'm not sure.  Responses from others welcome.

Reply:
I used StringBuilder because java strings are immutable and I would have to use substring method. About the tabs: With the approach, you suggested a counter for the number of tabs will be needed.